Allow commands_spec to be run against the installed copy of sugarjar

michel-slm opened the pull request:
For Ruby packages, Debian autopkgtest runs in an environment where `gem2deb-test-runner` removes the `lib` directory from the source tree, so the specs have to be able to load the installed copy instead.

With this change, `lib` gets added to the front of the path, and the relative `require` is changed to not specify a path. So when specs are run from a source tree checkout the `lib` in the checkout is still preferred, but if it's not available the installed library gets used to satisfy the `require`.

See https://wiki.debian.org/Teams/Ruby/Packaging/Tests#Case_eight:_autopkgtest_failure

Signed-off-by: Michel Alexandre Salim <michel@michel-slm.name>

jaymzh reviewed (dismissed):
I'm fine with this, especially since it's only in the specs. However, please add a significant comment in the code itself as to why it's this way instead of require_relative... basically the comment in your PR, lets just put it in the code.

michel-slm commented:
> I'm fine with this, especially since it's only in the specs. However, please add a significant comment in the code itself as to why it's this way instead of require_relative... basically the comment in your PR, lets just put it in the code.

makes sense! saves someone hunting the git blame for context.

jaymzh reviewed (changes requested):
You have some lint failures.

michel-slm commented:
Addressed the first one (style issue), the second one - the errors are a bit cryptic but I'm guessing it's because I use backticks in the comments. Removed those.

jaymzh commented:
Rebase for MDL fixes. Use rubocop's autocorrect for ruby fixes.
